### PR TITLE
Web game logic

### PR DESCRIPTION
**Pull request by vtodorchuk**

### CODE REVIEW

File: coverage/.last_run.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
```
Comment (CaptainJNS) [inline coverage/.last_run.json]: remove `coverage/`

File: spec/middlewares/routes_spec.rb [hunk 2]
```diff
@@ -0,0 +1,123 @@
+# frozen_string_literal: true
+
+RSpec.describe Middlewares::Routes do
+  include Rack::Test::Methods
+
+  def app
+    Rack::Builder.parse_file('config.ru').first
+  end
```
Comment (CaptainJNS) [inline spec/middlewares/routes_spec.rb:8]: do you need it?

File: spec/services/data_spec.rb [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  let(:table) { Entities::Rating.new }
+  let(:load) { Services::LoadService.new './tmp/data.yml' }
+  let(:save) { Services::SaveService.new table, './tmp/data.yml' }
+
+  describe '#execute' do
+    context 'when data save to file the same as read' do
+      before do
+        FileUtils.mkdir './tmp'
+
+        data = CodeBreaker::Game.new name: 'Vlad', difficulty: 'easy', secret_code: [1, 1, 1, 1]
```
Comment (CaptainJNS) [inline spec/services/data_spec.rb]: use `let`

File: spec/services/data_spec.rb [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+          name: data.name, difficulty: data.difficulty, total_attempts: data.total_attempts,
+          used_attempts: data.used_attempts, total_hints: data.total_hints,
+          used_hints: data.used_hints, date: Date.parse(DateTime.now.to_s)
+        }
+        save.save
+      end
+
+      after { FileUtils.remove_dir './tmp' }
+
+      it do
```
Comment (CaptainJNS) [inline spec/services/data_spec.rb]: describe tests

File: storage/data.yml [hunk 5]
```diff
@@ -0,0 +1,170 @@
+--- !ruby/object:Entities::Rating
```
Comment (CaptainJNS) [inline storage/data.yml]: to gitignore

**Review by CaptainJNS — CHANGES_REQUESTED** [review]

File: spec/middlewares/routes_spec.rb (hunk 2 above)
Comment (vtodorchuk) [reply spec/middlewares/routes_spec.rb:8]: Without this method test won`t pass. 

```
NameError:
        undefined local variable or method `app' for #<RSpec::ExampleGroups::MiddlewaresRoutes::WhenGameStart::WhenReceivedUserParams "redirects to game page" (./spec/middlewares/routes_spec.rb:118)>

```

**Review by CaptainJNS — APPROVED** [review]

**Comment by CaptainJNS** [comment]
Fix tests. But keep `data.yml` in gitignore

**Pull request merged**